contracts/YieldStrategyLever.sol

```diff
@@ -0,0 +1,155 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.14;
+
+import "./YieldLeverBase.sol";
+
+interface IStrategy {
+    function mint(address to) external returns (uint256 minted);
+
+    function burn(address to) external returns (uint256 withdrawal);
+
+    function burnForBase(address to) external returns (uint256 withdrawal);
+
+    function pool() external returns (IPool pool);
+}
+
+contract YieldStrategyLever is YieldLeverBase {
+    using TransferHelper for IERC20;
+    using TransferHelper for IFYToken;
+
+    mapping(bytes6 => IStrategy) strategies;
+    address constant USDC = 0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48;
+    address constant DAI = 0x6B175474E89094C44Da98b954EedeAC495271d0F;
+
+    constructor(Giver giver_) YieldLeverBase(giver_) {
+        IERC20(USDC).approve(
+            address(LADLE.joins(0x303200000000)),
+            type(uint256).max
+        );
+        IERC20(DAI).approve(
+            address(LADLE.joins(0x303100000000)),
+            type(uint256).max
+        );
+    }
+
+    function setStrategy(bytes6 ilkId, IStrategy strategy) external {
+        strategies[ilkId] = strategy;
+        IERC20(CAULDRON.assets(ilkId)).approve(
+            address(LADLE.joins(ilkId)),
+            type(uint256).max
+        );
+    }
+
+    /// @notice This function is called from within the flash loan.
+    /// @param ilkId The id of the ilk being borrowed.
+    /// @param seriesId The pool (and thereby series) to borrow from.
+    /// @param vaultId The vault id to put collateral into and borrow from.
+    /// @param borrowAmount The amount of FYTOKEN borrowed in the flash loan.
+    /// @param fee The fee that will be issued by the flash loan.
+    /// @param baseAmount The amount of own collateral to supply.
+    function borrow(
+        bytes6 ilkId,
+        bytes6 seriesId,
+        bytes12 vaultId,
+        uint128 baseAmount,
+        uint256 borrowAmount,
+        uint256 fee
+    ) internal override {
+        // We have borrowed FyTokens, so sell those
+        IPool pool = IPool(LADLE.pools(seriesId));
+        IFYToken tempFyToken = pool.fyToken();
+        tempFyToken.safeTransfer(address(pool), borrowAmount - fee);
+        uint256 baseReceived = pool.sellFYToken(address(pool), 0); // Sell fyToken to get USDC/DAI/ETH
+        pool.base().transfer(
+            address(pool),
+            pool.base().balanceOf(address(this))
+        );
+        // Mint LP token & deposit to strategy
+        pool.mintWithBase(
+            address(strategies[ilkId]),
+            msg.sender,
+            uint256(
+                pool.buyFYTokenPreview(uint128((baseReceived + baseAmount) / 3))
+            ), // TODO: what should be the fyTokenToBuy
```

Comment:
If we use `mintWithBase`, the caller needs to provide the `fyTokenToBuy` parameter.